### PR TITLE
Banana for Scale

### PR DESCRIPTION
**Pull request by mjcarroll**
# 🎉 Banana for Scale

## Summary

It is frequently difficult to communicate scale in a robotics simulation.  This plugin gives users the ability to insert bananas as a scale reference.  This adds two scales of bananas for scale: a normal banana and a big banana.  Users can insert the appropriate scale for their application.

The banana model is a photogrammetry reconstruction from @ColeOSRF.  The markings on the side give additional scale context.

banana for scale

## Test it

Start ignition gazebo and load the "Banana for Scale" plugin.  Insert bananas for scale.

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge**

### CODE REVIEW

**Comment by PeterMitrano** [comment]
image

File: src/gui/plugins/banana_for_scale/BananaForScale.cc [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+             fuelClient {nullptr};
+
+  };
+}
+
+const char kBanana[] =
+  "https://fuel.ignitionrobotics.org/1.0/mjcarroll/models/banana for scale";
+
+const char kBigBanana[] =
+  "https://fuel.ignitionrobotics.org/1.0/mjcarroll/models/big banana for scale";
```
Comment (nindoja) [inline src/gui/plugins/banana_for_scale/BananaForScale.cc:66]: Unencoded spaces in a URL. For shame!

**Review by caguero — COMMENTED** [review]
Does it make sense to add another two extra bananas in imperial units?

**Comment by darwinpanderson** [comment]
This is excellent, though I've a few clients with files that I need to support legacy measurements on. Any chance we might see the addition of an Imperial Plantain in the future along side your Metric Banana?

**Comment by chapulina** [comment]
> a normal banana and a big banana

Can you be more specific? How big is a "normal" banana? :thinking: 

image

**Comment by drewbrew** [comment]
How hard would it be to add a plantain for comparison? I could really go for some tostones.

File: src/gui/gui.config [hunk 2]
```diff
@@ -115,6 +115,19 @@
...
+    <property key="resizable" type="bool">false</property>
+    <property key="x" type="double">400</property>
+    <property key="y" type="double">0</property>
+    <property key="width" type="double">100</property>
+    <property key="height" type="double">50</property>
+    <property key="state" type="string">floating</property>
+    <property key="showTitleBar" type="bool">false</property>
+    <property key="cardBackground" type="string">#666666</property>
+  </ignition-gui>
+</plugin>
```
Comment (chapulina) [inline src/gui/gui.config:129]: As awesome as this plugin is, how about we leave it out of the default GUI configuration? Leave it as an Easter-egg that users can insert from the menu.

File: src/gui/plugins/banana_for_scale/BananaForScale.cc [hunk 3]
```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (C) 2020 Open Source Robotics Foundation
```
Comment (chapulina) [inline src/gui/plugins/banana_for_scale/BananaForScale.cc:2]:
```suggestion
 * Copyright (C) 2021 Open Source Robotics Foundation
```

File: src/gui/plugins/banana_for_scale/BananaForScale.cc [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+  class BananaPrivate
+  {
+    /// \brief Ignition communication node.
+    public: transport::Node node;
+
+    /// \brief Mutex to protect mode
+    public: std::mutex mutex;
+
+    /// \brief Transform control service name
+    public: std::string service;
```
Comment (chapulina) [inline src/gui/plugins/banana_for_scale/BananaForScale.cc:53]: These variables aren't used

File: src/gui/plugins/banana_for_scale/BananaForScale.cc [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+
+/////////////////////////////////////////////////
+void BananaForScale::LoadConfig(const tinyxml2::XMLElement *)
+{
+  if (this->title.empty())
+    this->title = "Banana for Scale";
+
+  // For shapes requests
+  ignition::gui::App()->findChild
+    <ignition::gui::MainWindow *>()->installEventFilter(this);
```
Comment (chapulina) [inline src/gui/plugins/banana_for_scale/BananaForScale.cc:91]: This event filter isn't needed

File: src/gui/plugins/banana_for_scale/BananaForScale.cc [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+#include <iostream>
+#include <string>
+
+#include <ignition/common/Console.hh>
+#include <ignition/gui/Application.hh>
+#include <ignition/gui/GuiEvents.hh>
+#include <ignition/gui/MainWindow.hh>
+#include <ignition/plugin/Register.hh>
+#include <ignition/transport/Node.hh>
+#include <ignition/transport/Publisher.hh>
```
Comment (chapulina) [inline src/gui/plugins/banana_for_scale/BananaForScale.cc:33]: Several of these includes can be removed

File: src/gui/plugins/banana_for_scale/BananaForScale.qml [hunk 7]
```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2020 Open Source Robotics Foundation
```
Comment (chapulina) [inline src/gui/plugins/banana_for_scale/BananaForScale.qml:2]:
```suggestion
 * Copyright (C) 2021 Open Source Robotics Foundation
```

**Review by chapulina — COMMENTED** [review]
Works great :banana:

**Review by chapulina — APPROVED** [review]
I took the liberty of slipping in some trivial changes in 6d85f6aa253bd293a2fa924d17259d890f2012ac. Let's get this in!

**Pull request merged**